File: src/main/java/org/launchcode/CoffeeShopDigital/model/Purchasable.java
package org.launchcode.CoffeeShopDigital.model;

public abstract class Purchasable {

    private String name = this.getClass().toString();
    private String description;
    private double price;
    private int quantity = 1;


    @Override
    public String toString() {
        return String.format("%1d\t%2s\t=\t$ %3f", getQuantity(), getName(), getPrice()*getQuantity());
    }


    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }
    public void setPrice(double price) {
        this.price = price;
    }

    public int getQuantity() {
        return quantity;
    }
    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}

File: src/main/angularUI/src/app/_models/purchasable.ts
export abstract class Purchasable {
    name!: any;
    price!: number;
    quantity!: number;
    description?: string = "";
    favorite?: boolean = false;
    stars?: number = 0;
    imageUrl?: string;


    constructor(aName: any, aPrice: number, anImage?: string) {
        this.name = aName;
        this.price = aPrice;
        this.imageUrl = anImage;
    }


    printOrderString(): string {
        return `${this.quantity}\t${this.name}\t=\t$ ${this.quantity * this.price}`;
    }

    setQuantity(newQty: number) {
        this.quantity = newQty;
    }

    getName(): any { return this.name; }
    getImage(): any { return this.imageUrl; }
}
